util/stump-nm/nm-api.lisp

```diff
@@ -0,0 +1,124 @@
+   (connection :initarg :connection :reader connection)
+   (type :initarg :type :reader cs-type)))
+
+(defun get-connection-settings (settings)
+  (mapcar (lambda (connection-path)
+            (block lambda
+              (let* ((connection (make-instance 'setting-connection
+                                                :path connection-path
+                                                :bus (bus settings)))
+                     (connection-settings (dbus-call connection "GetSettings")))
+                (dolist (setting connection-settings)
+                  (when (string= (first setting) "connection")
+                    (return-from lambda
+                      (let ((settings (second setting)))
+                        (make-instance
+                         'connection-settings
+                         :connection connection
+                         :id (first (assoc-value settings "id" :test #'string=))
+                         :uuid (first (assoc-value settings "uuid" :test #'string=))
+                         :type (first (assoc-value settings "type" :test #'string=))))))))))
+          (getprop settings "Connections")))
+
+(defclass wireless-device (dbus-object)
+  ((interface :initform "org.freedesktop.NetworkManager.Device.Wireless")))
+
+(defclass access-point (dbus-object)
+  ((interface :initform "org.freedesktop.NetworkManager.AccessPoint")
+   (active :initarg :active :reader active)
+   (ssid :accessor ssid)
+   (frequency :accessor frequency)
+   (strength :accessor strength)))
+
+(defun device-access-points (device)
+  (with-introspected-object (object (bus device) (path device) *service*)
+    (let ((active-access-point (getprop device "ActiveAccessPoint")))
+      (mapcar (lambda (access-point-path)
+                (prog1-let (ap (make-instance 'access-point
+                                              :path access-point-path
+                                              :bus (bus device)
+                                              :active (string= access-point-path
+                                                               active-access-point)))
+                  (setf (ssid ap) (access-point-ssid ap)
+                        (frequency ap) (getprop ap "Frequency")
+                        (strength ap) (getprop ap "Strength"))))
+              (object (interface device) "GetAllAccessPoints")))))
+
+(defun access-point-ssid (ap)
+  (format nil "~{~a~}"
+          (mapcar #'code-char
+                  (getprop ap "Ssid"))))
+
+(defclass active-connection (dbus-object)
+  ((interface :initform "org.freedesktop.NetworkManager.Connection.Active")
+   (uuid :accessor uuid)
+   (state :accessor state)))
+
+(defun get-active-connections (root)
+  (mapcar (lambda (connection-path)
+            (make-active-connection root connection-path))
+          (getprop root "ActiveConnections")))
+
+(defun make-active-connection (root connection-path)
+  (prog1-let (active-connection (make-instance
```

Comment:
If you don't the extra dependency you can use serapeum's lret for this https://github.com/ruricolist/serapeum/blob/master/REFERENCE.md#lret-rest-bindings-body-body

Reply:
I mind a bit, especially for such a big dependency.